test/execution_summary_test.py

```diff
@@ -1086,3 +1086,47 @@ class Bar(RunOnceTask):
         self.assertIn('Luigi Execution Summary', s)
         self.assertNotIn('00:00:00', s)
         self.assertNotIn('\n\n\n', s)
+
+    def test_statuse_with_task_retry(self):
```

Comment:
I have a few concerns regarding this test-case:

 * There's 3 tasks. Can we make this much easier and only have one, it's run twice (you can do `self.run_task(Foo())` twice if you want to emulate retry). All you need is a task that first fails and then succeeds. Right?
 * In the end of the test. Shouldn't you check that there are no failed tests?
 * Can you like the other tests also test that the string-output (the summary) contains the strings that we expect? Something like `self.assertIn('Luigi Execution Summary', s)`, `self.assertIn('failed but then succeeded in retry blah blah', s)`